darts/models/forecasting_model.py

```diff
@@ -101,6 +105,343 @@ def _build_forecast_series(self,
+
+        Returns
+        -------
+        forecast
+            A time series containing the forecast values for `target_series`, when successively applying the specified
+            model with the specified forecast horizon.
+        """
+        # handle case where target_series not specified
+        if target_series is None:
+            target_series = training_series
+
+        # construct predict kwargs dictionary
+        predict_kwargs = {}
+        if use_full_output_length is not None:
+            predict_kwargs['use_full_output_length'] = use_full_output_length
+
+        # construct fit function (used to ignore target series for univariate models)
+        if isinstance(self, MultivariateForecastingModel):
+            fit_function = self.fit
+        else:
+            fit_function = lambda train, target, **kwargs: self.fit(train, **kwargs)  # noqa: E731
+
+        # prepare the start parameter -> pd.Timestamp
+        if isinstance(start, float):
+            start_index = int((len(training_series.time_index()) - 1) * start)
+            start = training_series.time_index()[start_index]
+        elif isinstance(start, int):
+            start = training_series[start].time_index()[0]
+
+        # build the prediction times in advance (to be able to use tqdm)
+        if trim_to_series:
+            last_pred_time = training_series.time_index()[-forecast_horizon - stride]
+        else:
+            last_pred_time = training_series.time_index()[-stride - 1]
+
+        pred_times = [start]
+        while pred_times[-1] <= last_pred_time:
+            pred_times.append(pred_times[-1] + training_series.freq() * stride)
+
+        # iterate and predict pointwise
+        values = []
+        times = []
+
+        iterator = _build_tqdm_iterator(pred_times, verbose)
+
+        if not retrain and not self._fit_called:
+            fit_function(training_series.drop_after(start), target_series.drop_after(start), verbose=verbose)
+
+        for pred_time in iterator:
+            train = training_series.drop_after(pred_time)  # build the training series
+            target = target_series.drop_after(pred_time)  # build the target series
+            if (retrain):
+                fit_function(train, target)
+                pred = self.predict(forecast_horizon, **predict_kwargs)
+            else:
+                pred = self.predict(forecast_horizon, input_series=train, **predict_kwargs)
+            values.append(pred.values()[-1])  # store the N-th point
+            times.append(pred.end_time())  # store the N-th timestamp
+
+        forecast = TimeSeries.from_times_and_values(pd.DatetimeIndex(times), np.array(values))
+
+        return forecast
+
+    @classmethod
+    def gridsearch(model_class,
+                   parameters: dict,
+                   training_series: TimeSeries,
+                   target_series: Optional[TimeSeries] = None,
+                   forecast_horizon: Optional[int] = None,
```

Comment:
Could we support the same formats as for backtesting here?

Reply:
Are you referring to the data types of the function parameters? If yes, do you think for all parameters that are shared they should be the same? Because the reason for why `forecast_horizon` is optional in this case is that specifying it will lead to a expanding window evaluation, whereas leaving it set as `None` will require either the validation series arguments or the `use_fitted_values` parameter to be passed. That's why we decided not to set it to a default value of 1 here. But maybe you're referring to something else completely?

Reply:
Sorry, my comment was not super clear :)
I only meant that for the `forecast_horizon` parameter, it would be nice to support `float`, `int` and `pd.Timestamp` types (when it's not `None`), so that this parameter is treated the same as in the `backtest` method.

Reply:
I think you're talking about the `start` parameter of the `backtest` method, since `forecast_horizon` is also just expected to be an integer in `backtest`. The thing is, `gridsearch` does not take `start` as a parameter right now, instead it takes `num_predictions`, from which we derive start. Do you suggest we replace `num_predictions` by `start` and treat it like we do in `backtest`?

Reply:
Right, sorry! You are right I mixed up the two parameters, my bad. It would be nice though to find a way to make the two signatures more compatible, and replace `num_predictions` by `start` like you suggest.

Reply:
Done. We could make the `RegressionModel.backtest` function conform more as well by introducing the same `start` data types as for `ForecastingModel`. But this requires a bit more work since we need to change the checks, and we will probably want to reuse code from `ForecastingModel` to avoid duplication.